sklearn/_build_utils/__init__.py

```diff
@@ -71,6 +71,11 @@ def cythonize_extensions(extension):
         os.environ.get("SKLEARN_ENABLE_DEBUG_CYTHON_DIRECTIVES", "0") != "0"
     )
 
+    # Additional checks for Cython
+    generate_cython_annotations = (
+        os.environ.get("SKLEARN_GENERATE_CYTHON_ANNOTATIONS", "0") != "0"
```

Comment:
Will this generate annotations for every file? That feels more than one needs for development.

Currently I use an alias to run `cython` with our default directives:

```bash
alias cythonX="cython -X language_level=3 -X boundscheck=False -X wraparound=False -X initializedcheck=False -X nonecheck=False -X cdivision=True"
```

Concretely, I run the following for annotations:

```bash
cythonX -a sklearn/tree/_splitter.pyx
```